### PR TITLE
*: fix potential unreachable drop

### PR DESCRIPTION
**Pull request by BusyJay**
Unreachable report can be ignored if channel is full, which may lead to abandoned follower. This pr unifies `Unreachable` and `SnapshotStatus` as significant message and handle them separately from normal messages.

This is an alternative fix for issue #2341.

### CODE REVIEW

File: src/server/server.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver + 'static> Server<T, S> {
         region_split_size: usize,
         storage: Storage,
         raft_router: T,
-        snapshot_status_sender: Sender<SnapshotStatusMsg>,
+        significant_msg_sender: Sender<SignificantMsg>,
```
Comment (siddontang) [inline src/server/server.rs]: can we move the sender to the RaftRouter? I think we can only use raft router to communicate with raft store.

So the RaftRouter trait may add:

```
fn report_unreachable()
fn report_snapshot_status()
```

Comment (BusyJay) [reply src/server/server.rs]: Tried, but got a lot of unrelated changes, maybe in another pr.

**Comment by siddontang** [comment]
LGTM

File: src/server/server.rs (hunk 1 above)
Comment (siddontang) [reply src/server/server.rs]: Got it. I think it is worth to do it later.

**Comment by siddontang** [comment]
PTAL @overvenus

**Comment by siddontang** [comment]
PTAL @javaforfun

**Comment by javaforfun** [comment]
LGTM

**Pull request merged**